packages/sw-cli/src/lib/generate-sw.js

```diff
@@ -1,33 +1,70 @@
 const path = require('path');
 
-const getFileDetails = require('./utils/get-file-details');
-const filterFiles = require('./utils/filter-files');
 const copySWLib = require('./utils/copy-sw-lib');
-const generateGlobPattern = require('./utils/generate-glob-pattern');
+const getFileManifestEntries = require('./utils/get-file-manifest-entries');
 const writeServiceWorker = require('./write-sw');
+const errors = require('./errors');
 
 /**
- * Generate a working service worker with an inlined
+ * @example <caption>Generate a service worker for a project.</caption>
+ * const swCLI = require('sw-cli');
+ *
+ * swCLI.generateSW({
+ *   rootDirectory: './build/',
+ *   globPatterns: ['**\/*.{html,js,css}'],
+ *   globIgnores: ['admin.html']
+ *   serviceWorkerName: 'sw.js'
+ * })
+ * .then(() => {
+ *   console.log('Service worker generated.');
+ * });
+ *
+ * This method will generate a working Service Worker with an inlined
  * file manifest.
- * @return {Promise} The promise returned here will be used to exit the
- * node process cleanly or not.
+ * @param {Object} input
+ * @param {String} input.rootDirectory The root of the files you wish to
+ * be cached. This will also be the directory the service worker and library
+ * files are written to.
+ * @param {Array<String>} input.globPatterns Patterns to glob for when
+ * generating the build manifest.
+ * @param {String|Array<String>} input.globIgnores Add glob patterns that
+ * should exclude files from the build manifest.
```

Comment:
Always use parallel structures for descriptions of parameters. Specifically, this one should match `input.globPatterns` so:

'Patterns to exclude when generating the build manifest.'